Jenkinsfile

```diff
@@ -0,0 +1,112 @@
+                    checkout scm
+                }
+            }
+        }
+        stage('Full Build') {
+            options {
+                timeout(time: 1, unit: 'HOURS')
+            }
+            steps {
+                withMaven(maven: 'maven-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}') {
+                    sh 'mvn clean install -B $DISABLE_DOWNLOAD_PROGRESS_OPTS'
+                }
+            }
+        }
+
+        /*
+          Deploy stage will only be executed for deployable branches. These include master and any patch branch matching M.m.x format (i.e. 2.10.x, 2.9.x, etc...).
+          It will also only deploy in the presence of an environment variable JENKINS_ENV = 'prod'. This can be passed in globally from the jenkins master node settings.
+        */
+        stage('Deploy') {
+            when {
+                allOf {
+                    expression { env.CHANGE_ID == null }
+                    expression { env.BRANCH_NAME ==~ /((?:\d*\.)?\d*\.x|master)/ }
+                    environment name: 'JENKINS_ENV', value: 'prod'
+                }
+            }
+            steps{
+                withMaven(maven: 'maven-latest', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LINUX_MVN_RANDOM}') {
+                    sh 'mvn deploy -B -DskipStatic=true -DskipTests=true -DretryFailedDeploymentCount=10 $DISABLE_DOWNLOAD_PROGRESS_OPTS'
+                }
+            }
+        }
+    }
+    post {
+        always{
+            script {
+                postCommentIfPR("Build ${currentBuild.currentResult} See the job results in [legacy Jenkins UI](${BUILD_URL}) or in [Blue Ocean UI](${BUILD_URL}display/redirect).", "${GITHUB_USERNAME}", "${GITHUB_REPONAME}", "${GITHUB_TOKEN}")
+                if (currentBuild.currentResult == 'FAILURE') {
+                    def logUrl = env.BUILD_URL + 'consoleText'
+                    def response = sh(returnStdout: true, script: "curl -L -k ${logUrl}")
+                    def failureMessage = getFailureMessage(response)
+                    postCommentIfPR("${failureMessage}", "${GITHUB_USERNAME}", "${GITHUB_REPONAME}", "${GITHUB_TOKEN}")
+                }
+            }
+        }
+        success {
+            slackSend color: 'good', message: "SUCCESS: ${JOB_NAME} ${BUILD_NUMBER}"
+        }
+        failure {
+            slackSend color: '#ea0017', message: "FAILURE: ${JOB_NAME} ${BUILD_NUMBER}. See the results here: ${BUILD_URL}"
+        }
+        unstable {
+            slackSend color: '#ffb600', message: "UNSTABLE: ${JOB_NAME} ${BUILD_NUMBER}. See the results here: ${BUILD_URL}"
+        }
+        cleanup {
+            catchError(buildResult: null, stageResult: 'FAILURE') {
+                echo '...Cleaning up workspace'
+                cleanWs()
```

Comment:
❔ Do we still need to explicitly clean up the workspace on failures?

Reply:
removed